gms/impl/src/main/java/com/linkedin/metadata/resources/dashboard/GenericChartDownstreamLineageResource.java

```diff
@@ -0,0 +1,90 @@
+package com.linkedin.metadata.resources.dashboard;
+
+import com.linkedin.common.lineage.EntityRelationship;
+import com.linkedin.common.lineage.EntityRelationshipArray;
+import com.linkedin.common.lineage.GenericLineage;
+import com.linkedin.common.urn.ChartUrn;
+import com.linkedin.common.urn.DashboardUrn;
+import com.linkedin.dashboard.ChartKey;
+import com.linkedin.metadata.dao.BaseLocalDAO;
+import com.linkedin.metadata.dao.BaseQueryDAO;
+import com.linkedin.metadata.entity.ChartEntity;
+import com.linkedin.metadata.entity.DashboardEntity;
+import com.linkedin.metadata.query.CriterionArray;
+import com.linkedin.metadata.query.Filter;
+import com.linkedin.metadata.query.RelationshipDirection;
+import com.linkedin.metadata.relationship.Contains;
+import com.linkedin.metadata.restli.RestliUtils;
+import com.linkedin.parseq.Task;
+import com.linkedin.restli.common.ComplexResourceKey;
+import com.linkedin.restli.common.EmptyRecord;
+import com.linkedin.restli.server.PathKeys;
+import com.linkedin.restli.server.annotations.PathKeysParam;
+import com.linkedin.restli.server.annotations.RestLiCollection;
+import com.linkedin.restli.server.annotations.RestLiSimpleResource;
+import com.linkedin.restli.server.annotations.RestMethod;
+import com.linkedin.restli.server.resources.SimpleResourceTemplate;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.linkedin.metadata.dao.Neo4jUtil.createRelationshipFilter;
+import static com.linkedin.metadata.dao.utils.QueryUtils.newFilter;
+
+
+/**
+ * Rest.li entry point: /chart/{chartKey}/genericDownstreamLineage
+ */
+@RestLiSimpleResource(name = "genericDownstreamLineage", namespace = "com.linkedin.chart", parent = Charts.class)
+public final class GenericChartDownstreamLineageResource extends SimpleResourceTemplate<GenericLineage> {
+
+  private static final String CHART_KEY = Charts.class.getAnnotation(RestLiCollection.class).keyName();
+  private static final Filter EMPTY_FILTER = new Filter().setCriteria(new CriterionArray());
+  private static final Integer MAX_DOWNSTREAM_CNT = 100;
+
+  @Inject
+  @Named("dashboardDAO")
+  private BaseLocalDAO _dashboardDAO;
+
+  @Inject
+  @Named("datasetQueryDao")
+  private BaseQueryDAO _datasetQueryDAO;
+
+  public GenericChartDownstreamLineageResource() {
+    super();
+  }
+
+  @Nonnull
+  @RestMethod.Get
+  public Task<GenericLineage> get(@PathKeysParam @Nonnull PathKeys keys) {
+    final ChartUrn chartUrn = getUrn(keys);
+
+    return RestliUtils.toTask(() -> {
+      final List<DashboardUrn> downstreamDashboards = _datasetQueryDAO
```

Comment:
Should this use the _dashboardQueryDao_? Does this even exist?

Reply:
Or do we somehow use the Dataset DAO for this?

Reply:
i actually need to rename this variable- it is actually just `graphDBQueryDAO`

Reply:
Got it..